torch/csrc/distributed/rpc/rref_context.h

```diff
@@ -330,7 +334,11 @@ class TORCH_API RRefContext {
   // without confirmation is OK, because the creator would either call to_here
   // or forward the UserRRef, and both would then require confirmations from the
   // owner.
+#ifdef _WIN32
+ thread_local static bool recording_;
+#else
   static thread_local bool recording_;
+#endif
```

Comment:
What happens if we use `thread_local static` everywhere? And why does MSVC forbid the reverse order?